### PR TITLE
📝(doc) update joanie-connection documentation

### PR DESCRIPTION
**Pull request by jbpenrath**
## Purpose

We do not update joanie-connection.md documentation page since we rename the setting from `JOANIE` to `JOANIE_BACKEND`. Furthermore, we add new settings and the purpose of each of them was not explained.

## Proposal

- [x] Update `docs/joanie-connection.md`
- [x] Add missing entry into UPGRADE.md

### CODE REVIEW

File: docs/joanie-connection.md [hunk 1]
```diff
@@ -3,43 +3,115 @@ id: joanie-connection
...
+    "COURSE_RUN_SYNC_NO_UPDATE_FIELDS": [],
+    "DEFAULT_COURSE_RUN_SYNC_MODE": "sync_to_public",
 }
 ...
 ```
 
+As mentionned earlier, Joanie is treated as a LMS by Richie, so we had to bind Joanie settings with
+LMS backends settings. We achieve this by dynamically append `JOANIE_BACKEND` settings into 
+`RICHIE_LMS_BACKENDS` list if Joanie is enabled. To understand this point, you can take a look to
+the `post_setup` method of the Base class configuration into `settings.py`.
```
Comment (sampaccoud) [inline docs/joanie-connection.md:44]:
```suggestion
As mentionned earlier, Joanie is treated as a LMS by Richie, so we have to bind Joanie settings with
LMS backends settings. We achieve this by dynamically appending the `JOANIE_BACKEND` setting value into 
the `RICHIE_LMS_BACKENDS` list, if Joanie is enabled. To understand this point, you can take a look at
the `post_setup` method of the Base class configuration into `settings.py`.
```

File: docs/joanie-connection.md [hunk 2]
```diff
@@ -3,43 +3,115 @@ id: joanie-connection
...
+
+Here they are all settings available into `JOANIE_BACKEND`:
+
+### BASE_URL
+
+The base url on which the Joanie instance is hosted. This is used to construct the complete url of
+the API endpoint on which requests are made by Richie's frontend application. *Richie checks if this
+settings is settled to know if Joanie should be enabled or not.*
```
Comment (sampaccoud) [inline docs/joanie-connection.md:52]:
```suggestion
setting is set to know if Joanie should be enabled or not.*
```

**Comment by jbpenrath** [comment]
Oups !

File: docs/joanie-connection.md [hunk 3]
```diff
@@ -3,43 +3,115 @@ id: joanie-connection
...
+- Type: string
+- Required: No
+- Value: for example `r"^.*/api/v1.0/(course-runs|products)/([^/]*)/?$"`
+
+### COURSE_RUN_SYNC_NO_UPDATE_FIELDS
+
+Joanie resources (course runs and products) are all synchronized with Richie as a CourseRun. This
+setting is used to set the value of the `sync_mode` field when a course run is created on Richie.
+Read documentation of [the eponym `LMS_BACKENDS` settings](./lms-backends.md#default_course_run_sync_mode),
+to discover how it can be configured.
```
[the remark this answers is not in the document]
Comment (jbpenrath) [reply docs/joanie-connection.md]: Whoops!

**Review by sampaccoud — APPROVED** [review]

**Pull request merged**